<a href="https://colab.research.google.com/github/InnaTik/python_for_ds_task/blob/main/%D0%94%D0%97_6_%22hw_1_3_1_1_3_2_%D0%A6%D0%B8%D0%BA%D0%BB_for_ipynb%22.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Домашнє завдання до теми "Цикл for"

0. Розігрів. Напишіть цикл, який виводить на екран числа від 1 до 10 включно.

In [1]:
for i in range(1,11):
  print (i, end="; ")

1; 2; 3; 4; 5; 6; 7; 8; 9; 10; 

**Початок покладено!**

1. Задайте список `numbers`, який містить значення від 0 до 9 включно. Використовуючи цикл `for`, обчисліть суму елементів в створеному списку та виведіть фінальну суму на екран.

In [98]:
numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]

In [97]:
numbers = list(range(10)) # другий варіант

In [99]:
sum_numbers = 0
for number in numbers:
  sum_numbers += number

print(sum_numbers)

45


2. Використовуючи цикл  `for` виведіть для кожного елементу списку `numbers`,який ми створили в завданні 1, чи цей елемент дорівнює 5. Тобто після виконання циклу на екрані має бути 10 bool значень.

In [100]:
for number in numbers:
  print(number == 5, end="; ")

False; False; False; False; False; True; False; False; False; False; 

3. Використовуючи цикл  `for` для кожного елементу списку `numbers`, який ми створили в завданні 1, виявіть, чи цей елемент ділиться на 2 без остачі, та запишіть результат у новий список `numbers_divisible_by_2`.

Увага! Виконання завдання відрізняється від попереднього тим, що тепер ми результат обчислення на кожній ітерації записуємо у новий список. До того ж ми тут виконуємо іншу математичну операцію.

Виведіть фільнальний список `numbers_divisible_by_2` на екран.

Варіант 1:

In [16]:
for numbers_divisible_by_2 in numbers:
    print(numbers_divisible_by_2 % 2 == 0, end="; ")
# результат порівняння, але не список

True; False; True; False; True; False; True; False; True; False; 

Варіант 2:

In [17]:
numbers_divisible_by_2 = []

for number in numbers:
  numbers_divisible_by_2 = (number % 2 == 0)
  print(numbers_divisible_by_2, end="; ")

# по ідеї це теж результат порівняння і не список

True; False; True; False; True; False; True; False; True; False; 

Варіант 3:

In [23]:
numbers_divisible_by_2 = []

for number in numbers:
  #variable = (number % 2 == 0)  #ще один варіант
  #numbers_divisible_by_2.append(variable)
  numbers_divisible_by_2.append(number % 2 == 0)

print(numbers_divisible_by_2)

# це наче вже результат - список

[True, False, True, False, True, False, True, False, True, False]


Очікуваний результат: `[True, False, True, False, True, False, True, False, True, False]`

4. Задано два списки в клітинці нижче - `a` і `b`. Створіть цикл, результатом якого буде список `result`, що містить елементи типу bool і який фактично каже,
"чи на цьому індексі елементи наших двох списків співпадають". Приклад:
```
a = [1, 2, 3, 6]
b = [1, 7, 3, 6]
# ми виконали поелементне порівняння двох списків і отримали новий список result
# result містить наступний результат -> [True, False, True, True]
```
В кінці виведіть вміст result на екран.

<details>
<summary>Підказка</summary>

Аби порівняти елементи в циклі - треба створити цикл по індексам елементів. Зважаючи, що наші списки - одної довжини, цей підхід дозволить нам отримати чітко те, що треба.

**Просунутий варіант.** Альтернативно можна створити цикл, який на кожній ітерації має доступ до двох елементів - по одному з кожного списка, для цього треба скористатись функцією `zip` - приклад [тут](https://realpython.com/python-zip-function/#traversing-lists-in-parallel)
</details>


In [60]:
a = [64, 93, 60, 95, 66, 9, 15, 55, 77, 28, 74, 100, 13, 1, 85, 48, 66, 65, 59, 32]
b = [64, 93, 60, 95, 66, 9, 15, 56, 77, 28, 74, 100, 13, 1, 85, 48, 67, 65, 59, 32]

In [61]:
min_len = min(len(a),len(b))

result =[]

for i in range(min_len):
  if a[i] == b[i]:
    result.append(True)
  else:
    result.append(False)

print(result)

[True, True, True, True, True, True, True, False, True, True, True, True, True, True, True, True, False, True, True, True]


In [57]:
for index, (item1, item2) in enumerate(zip(a, b)):
    print({item1} == {item2}, end=", ")

True, True, True, True, True, True, True, False, True, True, True, True, True, True, True, True, False, True, True, True, 

5. Нижче задано список `numbers`. Виконайте клітинку з його заданням.

  Давайте знайдемо середнє арифметичне значення елементів списку `numbers` округлене до 2 цифр після коми та запишемо його у змінну `numbers_mean`. Для цього виконаємо два простих кроки.

  1) Знайдіть суму всіх елементів списку `numbers` та запишіть її у змінну `numbers_sum`. Можна з використанням циклу, а можна і без.

  2) Обчисліть `numbers_mean` шляхом ділення суми елементів у списку на їх кількість. Кількість елементів в списку - це його довжина.

  Виведіть `numbers_mean`, округлений до 2 цифр після коми, на екран.

  **Підказка**: округлити число можна з використанням вбудованої функції Python [round()](https://docs.python.org/3/library/functions.html#round).

In [63]:
numbers = [42, 51, 29, 2, 13, 44, 80, 66, 16]

In [73]:
numbers_sum = 0

for num in numbers:
    numbers_sum += num

numbers_mean = numbers_sum/len(numbers)

print("Avarege = ", round(numbers_mean,2))

Avarege =  38.11


Очікуваний результат: `38.11`

6. Задано список `a` і елемент `element`. Перевірте програмно чи є в списку `a` елемент `element` з допомогою інструкцій `if ... else`. Виведіть на екран відповідь "Так, є", якщо цей елемент в списку присутній, або "Ні, нема", якщо його там немає.
Важливо це зробити без використання імпортів будь-яких бібліотек.

  Тут цикла немає :) Але це завдання - підготовче до наступного.

In [80]:
a = [64, 93, 60, 95, 66, 9, 15, 55, 77, 28, 74, 100, 13, 1, 85, 48, 66, 65, 59, 32]
element = 33

In [79]:
if element in a:
  print("Так, є.")
else:
  print("Ні, нема.")

Ні, нема.


7. Тепер у нас не один елемент, а кілька. Вони наведені в змінній `elements`. За допомогою циклу і інструкцій `if ... else` для кожного елемента списку `elements` перевірте, чи він є в списку `a`.
Виведіть на екран відповідь "Так, є елемент <вставити значення елемента>", якщо елемент в списку присутній, або "Ні, нема елемента <вставити значення елемента>", якщо його там немає.

In [94]:
elements = [44, 66, 77, 100]

In [95]:
for element in elements:
  if element in a:
    print(f"Так, є елемент: {element}")
  else:
    print(f"Ні, нема елемента: {element}")

Ні, нема елемента: 44
Так, є елемент: 66
Так, є елемент: 77
Так, є елемент: 100
